R/linelist_merge.R

```diff
@@ -0,0 +1,416 @@
+      #replace everything that has a more than one comma or singula commas with no characters
+      recode_sheet<-data.frame(lapply(recode_sheet, function(x) {gsub("^,*|(?<=,),|,*$", "", x, perl=T) }), stringsAsFactors=F)
+      #replace blanks with NA
+      recode_sheet[recode_sheet==""]<-NA
+      #select newly recoded columns
+      recode_sheet$id<-rownames(recode_sheet) #create an id variable for when merging back to og sheet occurs
+      recode_sheet<- recode_sheet[ ,which(names(recode_sheet) %in% c(what_to_match,"id"))]
+
+      #remove the columns that required recoding from original sheet
+      output_sheet<-og_sheet[ , -which(names(og_sheet) %in% recode_vars)]
+
+
+      #match old variable names with dictionary for new variable names
+      names(output_sheet) <- with(var_dict_country,new_var[match(names(output_sheet),old_var_r)])
+      #cbind with newly recoded variables
+      output_sheet$id<-rownames(output_sheet)#create an id variable for when merging back to og sheet occurs
+      output_sheet<-merge(output_sheet,recode_sheet,by="id") %>% select(-c(id))
+
+      # if there are no variables that needed recoding just match old variable names with dictionary for new variable names
+      }else{
+        output_sheet<- og_sheet
+        #names(output_sheet) <- gsub(x= names(output_sheet),pattern ="\r\n",replacement = "", fixed = T)    #### maybe unneccesary so comment out
+        #names(output_sheet) <-gsub(x= names(output_sheet),pattern ="\\s+", replacement =" ")         #### maybe unneccesary so comment out
+         names(output_sheet) <- with(var_dict_country,new_var[match(names(output_sheet),old_var_r)])
+      }
+      }
+
+      #handle all date variables
+      require(anytime)
+    #handle dates that are numeric. All character dates -> NA.
+    # Convert to numeric and change date format using origin of excel
+    # Convert to character date for final merge
+    output_sheet$id<-rownames(output_sheet)
+
+    datesnumeric<-output_sheet %>% select(contains("Date", ignore.case = T),id) %>% mutate_all(.funs=gsub,pattern=".",replacement = "/", fixed=T, perl = T) %>%
+      mutate_at(vars(contains("Date", ignore.case = T)), as.numeric) %>%
+      mutate_at(vars(contains("Date", ignore.case = T)),as.Date, origin="1899-12-30") %>%
+      mutate_at(vars(contains("Date", ignore.case = T)), as.character)
+
+      #handle dates that are characters. Select first 10 characters in cell This will select all full dates and not any times appended.
+      # Convert character formats using parse date time.
+      # Convert to numeric and change date format using origin of excel
+      # Convert to character date for final merge
+      datescharacter<-output_sheet %>% select(contains("Date", ignore.case = T), id) %>%
+        mutate_at(vars(contains("Date", ignore.case = T)), lubridate::parse_date_time, orders=c("ymd","Ymd","dmy","dmY","%Y-%m-%d","%y-%m-%d","%d-%m-%y","dBY","ymd HMS","Ymd HMS")) %>%
```

Comment:
lubridate to imports if cant do with anytime (one or the other must be able to all of this no?)